classes/mail/Mailable.inc.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+/**
+ * @file classes/mail/Mailable.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Mailable
+ * @ingroup mail
+ *
+ * @brief Represents email message data
+ */
+
+namespace PKP\mail;
+
+use BadMethodCallException;
+use Illuminate\Mail\Mailable as IlluminateMailable;
+use InvalidArgumentException;
+use PKP\context\Context;
+use PKP\core\PKPApplication;
+use PKP\core\PKPContainer;
+use PKP\mail\variables\ContextEmailVariable;
+use PKP\mail\variables\QueuedPaymentEmailVariable;
+use PKP\mail\variables\RecipientEmailVariable;
+use PKP\mail\variables\ReviewAssignmentEmailVariable;
+use PKP\mail\variables\SenderEmailVariable;
+use PKP\mail\variables\SiteEmailVariable;
+use PKP\mail\variables\StageAssignmentEmailVariable;
+use PKP\mail\variables\SubmissionEmailVariable;
+use PKP\payment\QueuedPayment;
+use PKP\site\Site;
+use PKP\stageAssignment\StageAssignment;
+use PKP\submission\PKPSubmission;
+use PKP\submission\reviewAssignment\ReviewAssignment;
+use PKP\user\User;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionParameter;
+
+class Mailable extends IlluminateMailable
+{
+
+    /**
+     * Message object key in data array
+     * @var string
+     */
+    const DATA_KEY_MESSAGE = 'message';
+
+    /**
+     * Workflow stage ID associated with with email, const WORKFLOW_STAGE_ID_...
+     * @var string|null $stageId
+     */
+    protected $stageId;
+
+    /**
+     * @param array $args passed to the ancestors' constructor
+     */
+    public function __construct(array $args)
+    {
+        $this->setTemplateVariables($args);
+    }
+
+    /**
+     * @return string|null associated workflow stage ID
+     */
+    public function getStageId() : ?string
+    {
+        return $this->stageId;
+    }
+
+    /**
+     * Sets stage ID for this mailable
+     * @param int $stageId
+     * @return Mailable
+     */
+    public function stageId(int $stageId)
+    {
+        $this->stageId = $stageId;
+        return $this;
+    }
+
+    /**
+     * @param string $localeKey
+     * @throws BadMethodCallException
+     *
+     */
+    public function locale($localeKey)
+    {
+        throw new BadMethodCallException('This method isn\'t supported, data passed to ' . static::class .
+            ' should be already localized.');
+    }
+
+    /**
+     * Unlike Illuminate Mailable, accepts html string with variables
+     * @param string $view HTML string with template variables
+     * @param array $data variable => value; see also reserved keys DATA_KEY_...
+     * @return Mailable
+     */
+    public function view($view, array $data = [])
```

Comment:
Yes, you are right, it might not contain variables and it's inherited from Laravel. The difference between `html()` method is that it passes the string to the compiler. I vaguely recall we discussed leaving it but don't remember the specific use case.